evm/vm/message.py

```diff
@@ -187,12 +193,29 @@ def __init__(self,
             validate_access_list(access_list)
         self.access_list = access_list
 
+    @property
+    def paygas_total_fee(self):
+        return self.paygas_gasprice * self.transaction_gas_limit
+
+    @property
+    def paygas_gasprice(self):
+        if self._paygas_gasprice is not None:
+            return self._paygas_gasprice
+        else:
+            return 0
+
+    @paygas_gasprice.setter
+    def paygas_gasprice(self, gas_price):
```

Comment:
probably good to add validation for this value here to be **really** sure that it's valid.